scripts/helper.sh

```diff
@@ -38,3 +38,67 @@ function fail_if_error() {
     exit $retVal
   fi
 }
+
+function check_image_artifact_exists() {
+  local FILE=$1
+  if [[ ! -f "$FILE" ]]; then
+    echo "Error: Image artifact $FILE does not exist."
+    set_job_status "error"
+    exit 1
+  fi
+}
+
+# Set up the resolv.conf file for the image root
+# This is used to ensure that the image root can resolve DNS names during customization.
+function setup_resolv() {
+    # Copy the container's /etc/resolv.conf to the image root for customization
+    local IMAGE_ROOT_DIR=$1
+    local IMAGE_ROOT_RESOLV=$IMAGE_ROOT_DIR/etc/resolv.conf
+    local TMP_RESOLV=$IMAGE_ROOT_DIR/tmp/resolv.conf
+    if [ -f "$IMAGE_ROOT_RESOLV" -o -L "$IMAGE_ROOT_RESOLV" ]; then
+        mv "$IMAGE_ROOT_RESOLV" "$TMP_RESOLV"
+    fi
+    cp --remove-destination /etc/resolv.conf "$IMAGE_ROOT_RESOLV"
+}
+
+# Restore the image root's old /etc/resolv.conf after customization
+function restore_resolv() {
+    # Restore the image root's old /etc/resolv.conf after customization
+    local IMAGE_ROOT_DIR=$1
+    local IMAGE_ROOT_RESOLV=$IMAGE_ROOT_DIR/etc/resolv.conf
+    local TMP_RESOLV=$IMAGE_ROOT_DIR/tmp/resolv.conf
+    rm "$IMAGE_ROOT_RESOLV"
+    if [ -f "$TMP_RESOLV" -o -L "$TMP_RESOLV" ]; then
```

Comment:
Same concern here as with the test in the previous function.

Reply:
As with the previous function, we want to restore it no matter what it is.